pkg/cloud/instance.go

```diff
@@ -277,13 +277,15 @@ func (c *client) GetOrCreateVMInstance(
 		listVirtualMachineParams.SetName(csMachine.Name)
 		if listVirtualMachinesResponse, err2 := c.cs.VirtualMachine.ListVirtualMachines(listVirtualMachineParams); err2 == nil && listVirtualMachinesResponse.Count > 0 {
 			csMachine.Spec.InstanceID = pointer.StringPtr(listVirtualMachinesResponse.VirtualMachines[0].Id)
+			csMachine.Status.InstanceState = listVirtualMachinesResponse.VirtualMachines[0].State
 		} else {
 			c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(err2)
+			return err
 		}
```

Comment:
I prefer avoiding `else` statements wherever possible :) https://github.com/aws/eks-anywhere/blob/main/docs/developer/best-practice.md#returns

```suggestion
		if listVirtualMachinesResponse, err2 := c.cs.VirtualMachine.ListVirtualMachines(listVirtualMachineParams); err2 != nil || listVirtualMachinesResponse.Count == 0 {
			c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(err2)
			return err
		}
		csMachine.Spec.InstanceID = pointer.StringPtr(listVirtualMachinesResponse.VirtualMachines[0].Id)
		csMachine.Status.InstanceState = listVirtualMachinesResponse.VirtualMachines[0].State
```

Reply:
This is a good suggestion. But seems not working here. 
the `listVirtualMachinesResponse` is defined in if statement, and could not be used later on -- IDE is complaining `listVirtualMachinesResponse` is not defined in line 282